lgc/builder/llpcYCbCrAddressHandler.h

```diff
@@ -0,0 +1,88 @@
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+ /**
+  ***********************************************************************************************************************
+  * @file  llpcYCbCrAddressHandler.h
+  * @brief LLPC header file: contains the definition of LLPC class YCbCrAddressHandler.
+  ***********************************************************************************************************************
+  */
+
+#pragma once
+#include "llpcGfxRegHandler.h"
+#include "llpcBuilderImpl.h"
+
+namespace lgc
+{
+
+// =====================================================================================================================
+// The class is used for calculating and mantain the base address of each plane in YCbCr image.
+class YCbCrAddressHandler
+{
+public:
+    YCbCrAddressHandler(Builder* pBuilder, SqImgRsrcRegHandler* pSqImgRsrcRegHelper, GfxIpVersion* pGfxIpVersion)
+    {
+        m_pBuilder = pBuilder;
+        m_pRegHelper = pSqImgRsrcRegHelper;
+        m_pGfxIpVersion = pGfxIpVersion;
+        m_PlaneBaseAddresses.clear();
+        m_pOne = pBuilder->getInt32(1);
+    }
+
+    // Generate base address of each image plane
+    void GenBaseAddress(uint32_t planeNum);
+
+    // Generate height and pitch
+    void GenHeightAndPitch(uint32_t bits, uint32_t bpp, uint32_t xBitCount, bool isTileOptimal, uint32_t planeNum);
+
+    // Power2align operation
+    llvm::Value* Power2Align(llvm::Value* x, uint32_t align);
+
+    // Get specific plane
+    llvm::Value* GetPlane(uint32_t idx) { assert(idx < 3); return m_PlaneBaseAddresses[idx]; }
+
+    // Get pitch for Y plane
+    llvm::Value* GetPitchY() { return m_pPitchY; }
+
+    // Get pitch for Cb plane
+    llvm::Value* GetPitchCb() { return m_pPitchCb; }
+
+    // Get height for Y plane
+    llvm::Value* GetHeightY() { return m_pHeightY; }
+
+    // Get Height for Cb plane
+    llvm::Value* GetHeightCb() { return m_pHeightCb; }
+
+    SqImgRsrcRegHandler*               m_pRegHelper;
+    Builder*                           m_pBuilder;
+    llvm::SmallVector<llvm::Value*, 3> m_PlaneBaseAddresses;
+    llvm::Value*                       m_pPitchY;
+    llvm::Value*                       m_pHeightY;
+    llvm::Value*                       m_pPitchCb;
+    llvm::Value*                       m_pHeightCb;
+    llvm::Value*                       m_pOne;
+    GfxIpVersion*                      m_pGfxIpVersion;
```

Comment:
m_pGfxIpVersion -> m_pGfxIp
I didn't notice this in previous reviews. To keep concise and consistent with existing codes, better to name it gfxIp (not "Version") for this member.